### PR TITLE
Support reading iceberg tables on HDFS

### PR DESCRIPTION
**Pull request by ucasfl**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Support reading Iceberg tables on HDFS.

### Documentation entry for user-facing changes

- [x] Documentation is written (mandatory for new features)

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit 7900c3733523299ff765f7c8d026dc1d07e2aeb7 with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/70268/7900c3733523299ff765f7c8d026dc1d07e2aeb7/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/70268/7900c3733523299ff765f7c8d026dc1d07e2aeb7/style_check.html">❌ failure</a></td></tr>
<tbody>
</table>

File: tests/integration/test_storage_iceberg/test.py [hunk 1]
```diff
@@ -197,6 +201,16 @@ def get_creation_expression(
...
+    elif storage_type == "hdfs":
+        if table_function:
+            return f"""
+                icebergHDFS(hdfs, path = 'hdfs://hdfs1:9000/iceberg_data/default/{table_name}/', format={format})
+            """
+        else:
+            return f"""
+                DROP TABLE IF EXISTS {table_name};
+                CREATE TABLE {table_name}
+                ENGINE=IcebergLocal(local, path = 'hdfs://hdfs1:9000/iceberg_data/default/{table_name}/', format={format});"""
```
Comment (kssenii) [inline tests/integration/test_storage_iceberg/test.py:213]:
```suggestion
                ENGINE=IcebergHDFS(local, path = 'hdfs://hdfs1:9000/iceberg_data/default/{table_name}/', format={format});"""
```

File: tests/integration/helpers/hdfs_api.py [hunk 2]
```diff
@@ -254,6 +254,64 @@ def write_data(self, path, content):
...
+            allow_redirects=False,
+            headers={"host": str(self.hdfs_ip)},
+            params={"overwrite": "true"},
+            verify=False,
+            auth=self.kerberos_auth,
+        )
+
+        logging.debug("HDFS api response:{}".format(response.headers))
+
+        # additional_params = '&'.join(
```
Comment (kssenii) [inline tests/integration/helpers/hdfs_api.py:288]: remove?

**Review by kssenii — APPROVED** [review]

**Comment by kssenii** [comment]
> Integration tests (aarch64) [3/6] — fail: 40

`test_storage_iceberg`

**Comment by ucasfl** [comment]
> > Integration tests (aarch64) [3/6] — fail: 40
> 
> `test_storage_iceberg`

The error message says `Exception: Can't wait HDFS to start`, have no idea about it.

**Comment by ucasfl** [comment]
AST fuzzer(asan): #70393.

**Comment by kssenii** [comment]
It still fails

> Integration tests (aarch64) [3/6] — fail: 40

looks like the problem is related to aarch64 only

**Comment by ucasfl** [comment]
> It still fails
> 
> > Integration tests (aarch64) [3/6] — fail: 40
> 
> looks like the problem is related to aarch64 only

Looks like aarch64 also doesn't run tests with storage hdfs.

**Comment by kssenii** [comment]
> @kssenii Looks like aarch64 also doesn't run tests with storage hdfs.

then just disable these test with hdfs for aarch, it can be done inside the test itself

**Comment by ucasfl** [comment]
> > @kssenii Looks like aarch64 also doesn't run tests with storage hdfs.
> 
> then just disable these test with hdfs for aarch, it can be done inside the test itself

Any reference about it?

**Comment by kssenii** [comment]
https://github.com/ClickHouse/ClickHouse/blob/46824d04346bbab186bd372279f7c9374b0c9837/tests/integration/test_merge_tree_hdfs/test.py#L21-L22

**Comment by ucasfl** [comment]
> https://github.com/ClickHouse/ClickHouse/blob/46824d04346bbab186bd372279f7c9374b0c9837/tests/integration/test_merge_tree_hdfs/test.py#L21-L22

But looks like it will disable the whole test_storage_iceberg for aarch?

**Comment by kssenii** [comment]
The same can de done on test level, see https://github.com/ClickHouse/ClickHouse/blob/46824d04346bbab186bd372279f7c9374b0c9837/tests/integration/test_merge_tree_s3/test.py#L944-L945

**Review by kssenii — APPROVED** [review]

**Pull request merged**